tests/traits/GetTableColumnsTrait.php

```diff
@@ -15,13 +15,14 @@ trait GetTableColumnsTrait
      */
     public function getTableColumns(Connection $connection, array $table): array
     {
-        $sqlTemplate = 'select columnname from dbc.columns where tablename = %s and databasename = %s;';
+        $sqlTemplate = 'select columnname from dbc.columns where tablename = %s and databasename = %s order by %s;';
```

Comment:
seřadím sloupečky podle ColumnId - je to jen kvůli testům aby na výstupu byly sloupečky vždy ve stejným pořadí

Reply:
tady mi to hapávalo že si to v testech načetlo v jiném pořadí
https://github.com/keboola/teradata-transformation/actions/runs/3845245706/jobs/6549197637

Reply:
nechci ti do toho kecat, ale pouzivas spatne view. Toto view osekava nazvy na 30 znaku. Takto vypada jeho DDL
```
REPLACE VIEW DBC.Columns
  (DatabaseName,TableName,ColumnName,ColumnFormat,ColumnTitle,
   SPParameterType,ColumnType,ColumnUDTName,ColumnLength,DefaultValue,
   Nullable,CommentString,DecimalTotalDigits,DecimalFractionalDigits,
   ColumnId,UpperCaseFlag,Compressible,CompressValue, 
   ColumnConstraint, ConstraintCount, 
   CreatorName, CreateTimeStamp, LastAlterName, LastAlterTimeStamp,
   CharType, IdColType, AccessCount, LastAccessTimeStamp,
   CompressValueList, TimeDimension, VTCheckType, TTCheckType, ConstraintId,
   ArrayColNumberOfDimensions, ArrayColScope, ArrayColElementType, ArrayColElementUdtName, TSColumnType, RowVersionNo)
AS
SELECT (CAST(SUBSTRING(TRANSLATE(dbase.DatabaseName USING UNICODE_TO_LOCALE WITH ERROR) FROM 1 FOR 30)
             AS CHAR(30))),
       (CAST(SUBSTRING(TRANSLATE(tvm.TVMName USING UNICODE_TO_LOCALE WITH ERROR) FROM 1 FOR 30)
             AS CHAR(30))),
       (CAST(SUBSTRING(TRANSLATE(tvfields.FieldName USING UNICODE_TO_LOCALE WITH ERROR) FROM 1 FOR 30)
             AS CHAR(30))),
       (CAST(SUBSTRING(tvfields.FieldFormat FROM 1 FOR 30) AS CHAR(30))),
       (CAST(SUBSTRING(tvfields.FieldTitle FROM 1 FOR 60) AS VARCHAR(60))),
       tvfields.SPParameterType,
       tvfields.FieldType,
       (CAST(SUBSTRING(TRANSLATE(tvfields.UDTName USING UNICODE_TO_LOCALE WITH ERROR) FROM 1 FOR 30)
             AS CHAR(30))),
       tvfields.MaxLength(FORMAT 'Z,ZZZ,ZZZ,ZZ9'),    
       tvfields.DefaultValue,
       tvfields.Nullable,
       tvfields.CommentString,
       tvfields.TotalDigits(FORMAT '-ZZ9'),
       tvfields.ImpliedPoint(FORMAT '-ZZ9'),
       tvfields.FieldId,
       tvfields.UpperCaseFlag,  
       tvfields.Compressible,
       NULL,
       TRANSLATE(tvfields.ColumnCheck USING UNICODE_TO_LOCALE WITH ERROR), 
       tvfields.CheckCount,
       (CAST(SUBSTRING(TRANSLATE(DB1.DatabaseName USING UNICODE_TO_LOCALE WITH ERROR) FROM 1 FOR 30)
             AS CHAR(30))),
       tvfields.CreateTimeStamp,
       (CAST(SUBSTRING(TRANSLATE(DB2.DatabaseName USING UNICODE_TO_LOCALE WITH ERROR) FROM 1 FOR 30)
             AS CHAR(30))),
       tvfields.LastAlterTimeStamp,
       tvfields.CharType,
       tvfields.IdColType,
       OU.UserAccessCnt,
       OU.LastAccessTimeStamp,
       tvfields.CompressValueList,
       tvfields.TimeDimension,
       tvfields.VTCheckType,
       tvfields.TTCheckType,
       tvfields.ConstraintId,
       udt1.ArrayNumDimensions (NAMED ArrayColNumberOfDimensions),
       udt1.ArrayScope (NAMED ArrayColScope),
       tvf2.FieldType (NAMED ArrayColElementType),
       CAST(SUBSTRING(TRANSLATE(tvf2.UDTName USING UNICODE_TO_LOCALE WITH ERROR) FROM 1 FOR 30)
              AS CHAR(30)) (NAMED ArrayColElementUdtName),
       tvfields.TSColumnType (NAMED TSColumnType),
       tvfields.RowVersionNo
FROM DBC.tvfields
         LEFT OUTER JOIN DBC.ObjectUsage OU
                      ON OU.DatabaseId = DBC.TVFields.DatabaseId
                     AND OU.ObjectId = DBC.TVFields.TableId 
                     AND OU.FieldId = DBC.TVFields.FieldId
                     AND OU.IndexNumber IS NULL
                     AND OU.UsageType = 'DML'
         LEFT OUTER JOIN DBC.Dbase DB1
                      ON DBC.tvfields.CreateUID = DB1.DatabaseID
         LEFT OUTER JOIN DBC.Dbase DB2
                      ON DBC.tvfields.LastAlterUID = DB2.DatabaseID
         LEFT OUTER JOIN DBC.tvfields tvf2
                      ON DBC.tvfields.UDTypeId=tvf2.TableId AND 
                         DBC.tvf2.FieldName = 'ARRAYELEMENT' AND
           (DBC.tvfields.FieldType='A1' OR
                         DBC.tvfields.FieldType='AN')
         LEFT OUTER JOIN DBC.udtinfo udt1
                      ON DBC.tvfields.UDTypeId=udt1.TypeId AND (DBC.tvfields.FieldType='A1' OR
                                                                DBC.tvfields.FieldType='AN'),
          DBC.Dbase, DBC.TVM
WHERE   tvm.DatabaseId = dbase.DatabaseId  
        AND     tvm.tvmid = tvfields.tableid WITH CHECK OPTION;
        ```

pouzij `ColumnsVX`

Reply:
welcome, in teradata hell :D

Reply:
@jirkasemmler to je jen pro potřeby testu... není to nic co by ovlivnilo klienta

Reply:
jo aha, nevsiml jsem si, ze to je v testu. tak potom cajk

Reply:
pro manifest se načítají data jinak
https://github.com/keboola/teradata-transformation/blob/master/src/ManifestWriter.php#L53

Reply:
jo, v te reflexi je to cajk, tam to mame poresene